### PR TITLE
Update graphics contract, Update to latest version of BabylonNative

### PR DESCRIPTION
**Pull request by Drigax**
This PR integrates changes from https://github.com/BabylonJS/BabylonNative/pull/582, updating the BabylonNative graphics initialization code to consume the configuration-based contract.

### CODE REVIEW

File: Modules/@babylonjs/react-native/shared/BabylonNative.h [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace Babylon
 
     void Initialize(facebook::jsi::Runtime& jsiRuntime, Dispatcher jsDispatcher);
     void Deinitialize();
-    void UpdateView(void* windowPtr, size_t width, size_t height, void* windowTypePtr = nullptr);
+    void UpdateView(void* windowPtr, size_t width, size_t height);
```
Comment (Drigax) [inline Modules/@babylonjs/react-native/shared/BabylonNative.h:11]: Ideally, we should be removing `void *` from this header as well, in favor of WindowType.

Currently I cannot include the <Babylon> headers here without build failure, I'll keep tooling with CMake to so that this works, but it may require making this a library instead of a collection of source files included in another library. I'm no CMake expert, however.

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include "BabylonNative.h"
 
 #include <Babylon/Graphics.h>
+#include <Babylon/GraphicsPlatform.h>
```
Comment (Drigax) [inline Modules/@babylonjs/react-native/shared/BabylonNative.cpp:4]: After https://github.com/BabylonJS/BabylonNative/pull/656/files merges, we shouldn't need to include `GraphicsPlatform.h`

**Comment by chrisfromwork** [comment]
@Drigax in BabylonNative theres a change where openxr_loader.lib was renamed to openxr_loaderd.lib for debug. If you mimic what i did in the vcxproj here you should be able to fix windows ci: https://github.com/BabylonJS/BabylonReactNative/pull/188

File: Modules/@babylonjs/react-native/shared/CMakeLists.txt [hunk 3]
```diff
@@ -4,4 +4,4 @@ set(SHARED_INCLUDES
 set(SHARED_SOURCES
     "${CMAKE_CURRENT_LIST_DIR}/DispatchFunction.h"
     "${CMAKE_CURRENT_LIST_DIR}/BabylonNative.h"
-    "${CMAKE_CURRENT_LIST_DIR}/BabylonNative.cpp")
\ No newline at end of file
+    "${CMAKE_CURRENT_LIST_DIR}/BabylonNative.cpp")
```
Comment (chrisfromwork) [inline Modules/@babylonjs/react-native/shared/CMakeLists.txt:7]: nit: remove new line

File: Modules/@babylonjs/react-native/shared/BabylonNative.h (hunk 1 above)
Comment (chrisfromwork) [reply Modules/@babylonjs/react-native/shared/BabylonNative.h:11]: id suggest redefining the type here or using a void pointer for the time being. We can update include directories for the native modules to track down the headers required to access the window type definition. But right now the native modules aren't precompiled binaries when distributed through npm. So if BabylonNative.h references babylon native headers and our react-native native module references BabylonNative.h, i'm pretty sure wed have to package and distribute all headers required to resolve the windowtype, which is going to be a lot more work/seems beyond the scope of fixing breaking changes.

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp [hunk 4]
```diff
@@ -61,15 +62,20 @@ namespace Babylon
             Napi::Detach(m_env);
         }
 
-        void UpdateView(void* windowPtr, void* windowTypePtr, size_t width, size_t height)
+        void UpdateView(void* windowPtr, size_t width, size_t height)
         {
+            GraphicsConfiguration graphicsConfig = GraphicsConfiguration();
+            graphicsConfig.WindowPtr = (WindowType)windowPtr;
+            graphicsConfig.Width = width;
```
Comment (chrisfromwork) [inline Modules/@babylonjs/react-native/shared/BabylonNative.cpp:69]: nit can we have this as a reinterpret_cast<WindowType>()?

File: Modules/@babylonjs/react-native/android/src/main/cpp/BabylonNativeInterop.cpp [hunk 5]
```diff
@@ -66,7 +66,7 @@ extern "C" JNIEXPORT void JNICALL Java_com_babylonreactnative_BabylonNativeInter
 
 extern "C" JNIEXPORT void JNICALL Java_com_babylonreactnative_BabylonNativeInterop_00024BabylonNative_updateView(JNIEnv* env, jclass obj, jobject surface)
 {
-    ANativeWindow* windowPtr{ ANativeWindow_fromSurface(env, surface) };
+    ANativeWindow* windowPtr = ANativeWindow_fromSurface(env, surface);
     auto width{ static_cast<size_t>(ANativeWindow_getWidth(windowPtr)) };
```
Comment (chrisfromwork) [inline Modules/@babylonjs/react-native/android/src/main/cpp/BabylonNativeInterop.cpp:70]: nit: was this change needed?

**Review by chrisfromwork — APPROVED** [review]
:shipit:

File: Modules/@babylonjs/react-native/shared/BabylonNative.h (hunk 1 above)
Comment (Drigax) [reply Modules/@babylonjs/react-native/shared/BabylonNative.h:11]: After chatting offline, I'll go forward with retaining void*.

File: Modules/@babylonjs/react-native/android/src/main/cpp/BabylonNativeInterop.cpp (hunk 5 above)
Comment (Drigax) [reply Modules/@babylonjs/react-native/android/src/main/cpp/BabylonNativeInterop.cpp:70]: not at all, reverting

**Comment by Drigax** [comment]
Looks like everything's green outside of the Android build failures

File: Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/BabylonReactNative.vcxproj [hunk 6]
```diff
@@ -192,7 +192,7 @@
         NativeXr.lib;
         nvtt.lib;
         OGLCompilerd.lib;
-        openxr_loader.lib;
+        openxr_loaderd.lib;
```
Comment (ryantrem) [inline Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/BabylonReactNative.vcxproj:195]: Aren't these changes part of @chrisfromwork's PR? Not sure why they are showing up here.

Comment (Drigax) [reply Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/BabylonReactNative.vcxproj:195]: These changes were recommended by chris, I added them to this PR so that it would pass windows CI. We can check in #188 and I can rebase, else we pretty much have the same goal here, I'm just integrating my changes from the new latest commit.

Comment (chrisfromwork) [reply Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/BabylonReactNative.vcxproj:195]: yeah in moving to the newest master we'll see breaks without this change

**Comment by Drigax** [comment]
@ryantrem, are we good to check in?

**Review by ryantrem — APPROVED** [review]

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp (hunk 2 above)
Comment (Drigax) [reply Modules/@babylonjs/react-native/shared/BabylonNative.cpp]: Removed.

File: Modules/@babylonjs/react-native/shared/BabylonNative.cpp [hunk 7]
```diff
@@ -61,15 +61,20 @@ namespace Babylon
             Napi::Detach(m_env);
         }
 
-        void UpdateView(void* windowPtr, void* windowTypePtr, size_t width, size_t height)
+        void UpdateView(void* windowPtr, size_t width, size_t height)
         {
+            GraphicsConfiguration graphicsConfig = GraphicsConfiguration();
```
Comment (ryantrem) [inline Modules/@babylonjs/react-native/shared/BabylonNative.cpp:66]:
```suggestion
            GraphicsConfiguration graphicsConfig{};
```
